WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift

```diff
@@ -283,6 +286,48 @@ extension WPRichContentView: WPTextAttachmentManagerDelegate {
         return img
     }
 
+    func galleryForAttachment(_ attachment: WPTextAttachment) -> WPRichTextGallery {
+
+        let width: CGFloat = attachment.width > 0 ? attachment.width : self.bounds.width
+        let height: CGFloat = Constants.defaultGalleryHeight
+
+        let gallery = WPRichTextGallery(frame: CGRect(x: 0.0, y: 0.0, width: width, height: height))
+
+        attachment.maxSize = CGSize(width: width, height: height)
+
+        gallery.handleGalleryTapped = handleGalleryTapped
+        gallery.isPrivate = isPrivate
+
+        if let htmlString = attachment.html, htmlString.count > 0 {
+
+            var content = htmlString
+
+            //remove gallery tags
+            let galleryTagsStart = try! NSRegularExpression(pattern: "<gallery[^>]*>", options: .caseInsensitive)
+            let galleryTagsEnd = try! NSRegularExpression(pattern: "</gallery>", options: .caseInsensitive)
+
+            content = galleryTagsStart.stringByReplacingMatches(in: content,
+                                                                options: .reportCompletion,
+                                                                range: NSRange(location: 0, length: content.count),
+                                                                withTemplate: "")
+
+            content = galleryTagsEnd.stringByReplacingMatches(in: content,
+                                                                options: .reportCompletion,
+                                                                range: NSRange(location: 0, length: content.count),
+                                                                withTemplate: "")
+
+            //get images as their own attachments, to become datasource
+            let formatter = WPRichTextFormatter()
+            let (_, attachments) = formatter.processAndExtractTags(content)
+
+            gallery.imageAttachments = attachments.filter({ (attachment) -> Bool in
```

Comment:
You could condense this if you like, to `attachments.filter({ $0.tagName == "img" })`